source/euler/hyperbolic_system.h

```diff
@@ -1313,5 +1339,159 @@ namespace ryujin
+
+
+       //some general variables
+       dealii::Point<dim> disk_center;//center at 0,0 for every case, how to not hardcode this?
+       //FIXME: how do we calculate disk center from the information in mesh generation?
+       double disk_radius = 0.25;
+
+       //first, set up the finite element, the data, and the facevalues
+       dealii::FE_Q<dim> fe(ORDER_FINITE_ELEMENT);//the finite element
+       const int degree = fe.degree;
+       dealii::QGauss<dim-1> face_quadrature_formula(degree + 2);
+       const int n_q_points = face_quadrature_formula.size();
+
+       std::vector<double>      pressure_values(n_q_points);
+
+       scalar_type density, pressure;
+       std::vector<scalar_type> momentum(dim);
+
+   //    std::cout << "inside calc before density partition" << std::endl;
+       //initialize partitions
+       density.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       pressure.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       for(unsigned int c=0; c < dim; c++)
+         momentum.at(c).reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+
+       dealii::Tensor<1,dim> normal_vector;
+       dealii::SymmetricTensor<2,dim> fluid_stress;
+       dealii::SymmetricTensor<2,dim> fluid_pressure;
+       dealii::Tensor<1,dim> forces;
+
+       dealii::FEFaceValues<dim> fe_face_values(fe, face_quadrature_formula,
+           dealii::update_values | dealii::update_quadrature_points | dealii::update_gradients |
+           dealii::update_JxW_values | dealii::update_normal_vectors); //the face values
+
+       double drag = 0.;
+       double lift = 0.;
+
+
+       // Create vectors that store the locally owned parts on every process
+       U.extract_component(density, 0);//extract density
+       U.extract_component(pressure, dim+1);//extract density
+
+       //extract momentum, and convert to velocity
+       for(int c=0; c<dim; c++)
+       {
+         int comp = c+1;//momentum is stored in positions [1,...,dim], so add one to c
+         U.extract_component(momentum.at(c), comp);
+       }
+
+       //extract energy
+       U.extract_component(pressure, dim+1);
+
+       //convert E to pressure
+       for (unsigned int k=0; k<offline_data_.n_locally_owned(); k++)
+       {
+
+         //calculate momentum norm squared
+         const double &E = pressure.local_element(k);
+         const double &rho = density.local_element(k);
+         double m_square = 0;
+         for(int d=0; d<dim; d++)
+           m_square += std::pow(momentum.at(d).local_element(k),2);
+
+         //pressure = (gamma-1)*internal_energy
+         pressure.local_element(k) =  (gamma() - 1.0) * (E - 0.5*m_square/rho);
+       }
+
+
+       if((pressure.has_ghost_elements() != true) || (density.has_ghost_elements() != true)
+           ||(momentum.at(0).has_ghost_elements() != true) || (momentum.at(1).has_ghost_elements() != true))
```

Comment:
The condition asks whether any of the vectors does not have ghost elements. That's hard to read. It could probably use a comment. Either way, what you have here is an *assertion*: You want to check that something is true, and if it is not, you want to abort the program. That's exactly what the `Assert` statement is about.

Reply:
TODO: change this into a ASSERT